numpy/lib/_iotools.py

```diff
@@ -180,7 +181,7 @@ def __init__(self, delimiter=None, comments='#', autostrip=True,
         elif hasattr(delimiter, '__iter__'):
             _handyman = self._variablewidth_splitter
             idx = np.cumsum([0] + list(delimiter))
-            delimiter = [slice(i, j) for (i, j) in zip(idx[:-1], idx[1:])]
+            delimiter = [slice(i, j) for (i, j) in itertools.pairwise(idx)]
```

Comment:
I suppose we might as well go all the way then

```suggestion
            delimiter = list(itertools.starmap(slice, itertools.pairwise(idx)))
```

Reply:
That would be enforced by [FURB140](https://docs.astral.sh/ruff/rules/reimplemented-starmap/#reimplemented-starmap-furb140). I haven't applied that rule yet. By experience, maintainers in some projects don't like this rule becaus ethey find the code more complex.

Is it OK to leave this as is for now, and apply [FURB140](https://docs.astral.sh/ruff/rules/reimplemented-starmap/#reimplemented-starmap-furb140) later on?

Reply:
Yea sure; that's probably the path of least resistance